### PR TITLE
Refactor metadata structure

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
Changes the allowed structure of metadata to 
```
"metadata": [
    {
        "name": "period_id",
        "validator": "string"
    },
```
from
```
"metadata": {
    "period_id": {
        "validator": "string"
    },
```

### How to review 
This branch points to validator change in https://github.com/ONSdigital/eq-schema-validator/pull/70

### CODE REVIEW

**Comment by benhowes** [comment]
@jonnyshaw89 do all the tests pass locally for you? When I checkout this branch locally I see a lot of test fails relating to `sdc.crypto.exceptions.InvalidTokenException: Missing key/value for tx_id`

**Comment by jonnyshaw89** [comment]
@benhowes they did last time i ran them

**Comment by jonnyshaw89** [comment]
@benhowes its because i removed the `tx_id` generator.
i'm updating them now

**Review by MebinAbraham — COMMENTED** [review]
Personal opinion but I think these should have been 2 PRs, one to change the metadata structure and one to change the JSON structure.

Also FYI some schema still using the old metadata format.

**Comment by jonnyshaw89** [comment]
@MebinAbraham it would take me hours and hours to go through each schema and only change the metadata.

**Comment by MebinAbraham** [comment]
@jonnyshaw89 Fair.

File: data/cy/test_language.json [hunk 1]
```diff
@@ -1,32 +1,25 @@
 {
-    "mime_type": "application/json/ons/eq",
```
Comment (kierenbeckett) [inline data/cy/test_language.json]: Ideally we'd keep these at the top of each schema

**Review by kierenbeckett — CHANGES_REQUESTED** [review]
Perhaps rather than attempt to review the changes to the schemas, you could paste the script you used to reformat them?

**Comment by kierenbeckett** [comment]
You could change the JSON structure first, then change the metadata?

**Comment by jonnyshaw89** [comment]
@kierenbeckett the script to format is here https://github.com/ONSdigital/eq-survey-runner/pull/1697/files#diff-ba210a9157252cc983268fdc3aa3ed52R194

File: gulpfile.babel.js [hunk 2]
```diff
@@ -189,6 +191,7 @@ gulp.task('lint:json', () => {
 gulp.task('format:json', () => {
   return gulp
     .src(['./data/*/*.json'])
+    .pipe(sortJSON({ space: 2 }))
```
Comment (kierenbeckett) [inline gulpfile.babel.js]: Can we update `.editorconfig` to match

**Review by kierenbeckett — DISMISSED** [review]

**Comment by kierenbeckett** [comment]
How did you reformat the metadata?

**Comment by jonnyshaw89** [comment]
@kierenbeckett i knocked together a script. does that need to be part of this PR?

**Review by MebinAbraham — COMMENTED** [review]
Validator is only checking `data/en` atm, we should change sometime.

`data/cy/test_language.json`
```
"metadata": {
        "period_id": {
            "validator": "string"
        },
        "ru_name": {
            "validator": "string"
        },
        "user_id": {
            "validator": "string"
        }
    },
```

**Comment by kierenbeckett** [comment]
Doesn't need to be in the PR, could you paste it in a comment/gist?  I can't realistically review the schema diffs but I could at least check the method used to rewrite them.

**Comment by jonnyshaw89** [comment]
@kierenbeckett here is the script used
https://gist.github.com/jonnyshaw89/803b9a95c48a94468ca9f10a5b45cba5

**Review by kierenbeckett — CHANGES_REQUESTED** [review]
Would be good if you can find a way to specify the ordering or the fields in the JSON or failing that at least preserve the existing ordering.  Otherwise LGTM.

**Review by kierenbeckett — APPROVED** [review]

**Review by insacuri — APPROVED** [review]
Code changes look reasonable, linting and tests pass to LGTM

**Pull request merged**